config/config_file.go

```diff
@@ -58,3 +50,115 @@ func (l KnownConfigLoader) Configure(cfg *Config) error {
 	}
 	return nil
 }
+
+func (l KnownConfigLoader) resolveConfigFile(cfg *Config) (string, error) {
+	configFile := cfg.ConfigFile
+	if configFile == "" {
+		configFile = "~/.databrickscfg"
+	}
+	if strings.HasPrefix(configFile, "~") {
+		homedir, err := os.UserHomeDir()
+		if err != nil {
+			return "", fmt.Errorf("cannot find homedir: %w", err)
+		}
+		configFile = fmt.Sprintf("%s%s", homedir, configFile[1:])
+	}
+	_, err := os.Stat(configFile)
+	if err != nil {
+		return "", err
+	}
+	return configFile, nil
+}
+
+// ImplicitProfileLoader is an alternative config loader, which triggers if you
+// specify  `host`, but no other credentials either through direct configuration
+// or through environment variables, Databricks SDK for Go will try picking up
+// profile with the matching host from ~/.databrickscfg. This allows keeping
+// the hostname checked in to version control, but have ability to pick up
+// different credentials either from local development machine or production
+// server.
+type ImplicitProfileLoader struct {
+	KnownConfigLoader
+}
+
+func (l ImplicitProfileLoader) Name() string {
+	return "implicit-profile"
+}
+
+func (l ImplicitProfileLoader) resolveSection(
+	cfg *Config, iniFile *ini.File) (*ini.Section, error) {
+	var candidates []*ini.Section
+	for _, section := range iniFile.Sections() {
+		hash := section.KeysHash()
+		host, ok := hash["host"]
+		if !ok {
+			// if host is not set
+			continue
+		}
+		canonical, err := canonicalHost(host)
+		if err != nil {
+			// we're fine with other corrupt profiles
+			continue
+		}
+		if canonical != cfg.Host {
+			continue
+		}
+		candidates = append(candidates, section)
+	}
+	if len(candidates) == 0 && iniFile.HasSection("DEFAULT") {
+		return iniFile.Section("DEFAULT"), nil
+	}
+	if len(candidates) == 0 {
+		return nil, ErrCannotConfigureAuth
+	}
+	// in the real situations, we don't expect this to happen often
+	// (if not at all), hence we don't trim the list
+	if len(candidates) > 1 {
+		var profiles []string
+		for _, v := range candidates {
+			profiles = append(profiles, v.Name())
+		}
+		return nil, fmt.Errorf("%s match %s in %s",
+			strings.Join(profiles, " and "), cfg.Host, cfg.ConfigFile)
+	}
+	return candidates[0], nil
+}
+
+func (l ImplicitProfileLoader) Configure(cfg *Config) error {
+	if cfg.Host == "" || cfg.Profile != "" {
+		// no host to match or profile already configured
```

Comment:
Before running this we should try to configure the auth type and proceed only if this fails (nothing configured).